pandas/_libs/index_class_helper.pxi.in

```diff
@@ -77,4 +77,14 @@ cdef class {{name}}Engine(IndexEngine):
 
         return self._unpack_bool_indexer(indexer, val)
 
+    {{if name in {'Float64', 'Float32'} }}
+    def get_stargets(self, ndarray targets):
+        stargets = set(targets)
+        # account for nans
+        if -1 not in stargets and np.isnan(targets).any():
```

Comment:
we can improve on this slightly by first doing a (cached) check for if any of the engine's values are nan.  if not, then we dont need to adjust stargets